### ISSUE TITLE
`strengthen` is not safe

### ISSUE THREAD

**Issue by raymond-h** [nikita-volkov/refined#71]
`strengthen` allows you to fabricate refinements that are incorrect:
```hs
n = $$(refineTH @Positive 5)

n' :: Refined (Positive && Even) Integer
n' = strengthen n

n'' :: Refined Even Integer
n'' = andRight n'

print n''
```

Output:
```
n :: Refined Positive Integer
n' :: Refined (Positive && Even) Integer
n'' :: Refined Even Integer
Refined 5
```

Nothing is wrong with the `Even` predicate as it does correctly catch the error when not using `strengthen`.

This was tested with `refined-0.6.1`.

**Comment by chessai** [comment]
I could've sworn I fixed this already.

Fix incoming

**Comment by chessai** [comment]
corrected in efa5aaad04da2334f138d1ac70e342cc3a2d0975

**Comment by chessai** [comment]
released at https://hackage.haskell.org/package/refined-0.6.2
